api/server/server.go

```diff
@@ -1290,27 +1306,28 @@ func createRouter(eng *engine.Engine, logging, enableCors bool, dockerVersion st
 			"/exec/{id:.*}/json":              getExecByID,
 		},
 		"POST": {
-			"/auth":                         postAuth,
-			"/commit":                       postCommit,
-			"/build":                        postBuild,
-			"/images/create":                postImagesCreate,
-			"/images/load":                  postImagesLoad,
-			"/images/{name:.*}/push":        postImagesPush,
-			"/images/{name:.*}/tag":         postImagesTag,
-			"/containers/create":            postContainersCreate,
-			"/containers/{name:.*}/kill":    postContainersKill,
-			"/containers/{name:.*}/pause":   postContainersPause,
-			"/containers/{name:.*}/unpause": postContainersUnpause,
-			"/containers/{name:.*}/restart": postContainersRestart,
-			"/containers/{name:.*}/start":   postContainersStart,
-			"/containers/{name:.*}/stop":    postContainersStop,
-			"/containers/{name:.*}/wait":    postContainersWait,
-			"/containers/{name:.*}/resize":  postContainersResize,
-			"/containers/{name:.*}/attach":  postContainersAttach,
-			"/containers/{name:.*}/copy":    postContainersCopy,
-			"/containers/{name:.*}/exec":    postContainerExecCreate,
-			"/exec/{name:.*}/start":         postContainerExecStart,
-			"/exec/{name:.*}/resize":        postContainerExecResize,
+			"/auth":                                     postAuth,
+			"/commit":                                   postCommit,
+			"/build":                                    postBuild,
+			"/images/create":                            postImagesCreate,
+			"/images/load":                              postImagesLoad,
+			"/images/{name:.*}/push":                    postImagesPush,
+			"/images/{name:.*}/tag":                     postImagesTag,
+			"/containers/create":                        postContainersCreate,
+			"/containers/{name:.*}/kill":                postContainersKill,
+			"/containers/{name:.*}/pause":               postContainersPause,
+			"/containers/{name:.*}/unpause":             postContainersUnpause,
+			"/containers/{name:.*}/restart":             postContainersRestart,
+			"/containers/{name:.*}/start":               postContainersStart,
+			"/containers/{name:.*}/stop":                postContainersStop,
+			"/containers/{name:.*}/wait":                postContainersWait,
+			"/containers/{name:.*}/resize":              postContainersResize,
+			"/containers/{name:.*}/attach":              postContainersAttach,
+			"/containers/{name:.*}/copy":                postContainersCopy,
+			"/containers/{name:.*}/exec":                postContainerExecCreate,
+			"/exec/{name:.*}/start":                     postContainerExecStart,
+			"/exec/{name:.*}/resize":                    postContainerExecResize,
+			"/containers/{name:.*}/rename/{newname:.*}": postContainerRename,
```

Comment:
^^ perhaps a more generic endpoint (.e.g. `/containers/{name:*}/modify`). If other properties of containers are allowed to be updated in future, the same endpoint could be reused.

Also, because this is updating a container, strictly it should be `PUT` or `PATCH`, not `POST`?